pytest_operator/plugin.py

```diff
@@ -1091,58 +1099,120 @@ async def download_resources(
 
     async def build_bundle(
         self,
-        bundle: Optional[str] = None,
-        output_bundle: Optional[str] = None,
-        serial: bool = False,
-    ):
-        """Builds bundle using juju-bundle build."""
-        cmd = ["juju-bundle", "build"]
-        if bundle is not None:
-            cmd += ["--bundle", bundle]
-        if output_bundle is not None:
-            cmd += ["--output-bundle", output_bundle]
-        if self.destructive_mode:
-            cmd += ["--destructive-mode"]
-        if serial:
-            cmd += ["--serial"]
-        await self.run(*cmd, check=True)
+        bundle: Union[str, os.PathLike],
```

Comment:
I'm not actually convinced that the signature actually needs to change. The method signature and docstrings of an API essentially define the contract for method itself. The existing contract is fairly vague in the specification, but clear in the parameters. This change can largely (if not wholly) still honor the contract of the method without making a breaking API change. An additional advantage/improvement would be to further refine and clarify the method contract in the docstrings.

As there are existing users of this library, if the API is changed then it will break a downstream consumer of the library. Since the method contract can be upheld with the existing signature, it seems better to me to *not* change the signature and subsequently not break downstream consumers.

Just my $0.02